pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/ca/PooledConnectionFailoverIntegrationTest.java

```diff
@@ -50,6 +50,8 @@ public void test4_1_pooledWriterConnection_basicfailover()
 
     // Assert that the pooled connection is valid.
     assertTrue(testConnection.isValid(IS_VALID_TIMEOUT));
+
+    testConnection.close();
```

Comment:
Might be worth investigating why the testConnection.close() in reviveInstancesAndCloseTestConnection() doesn't affect the flakiness.